### PR TITLE
add Multi-cluster Replication docs

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

add Multi-cluster Replication docs

https://deploy-preview-480--mystifying-fermi-1bc096.netlify.app/docs/server/multi-cluster/

## Notes to reviewers

porting over https://github.com/uber/cadence/blob/master/docs/design/2290-cadence-ndc.md

[we agreed internally](https://temporaltechnologies.slack.com/archives/C018F52DUKU/p1622669725018900) to call it Multi-Cluster going forward to minimize the emphasis on datacenter

### CODE REVIEW

File: docs/server/multi-cluster.md [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+This guide introduces Temporal's Multi-cluster Replication capabilities.
+You can set this up with [`clusterMetadata` configuration](/docs/server/configuration#clustermetadata), however this is only meant to be a conceptual guide rather than a detailed tutorial.
+Please reach out to us if you need to set this up.
+
+## Overview
+
+Multi-cluster Replication is a feature which asynchronously replicates workflows from active cluster to other passive clusters, for backup & state reconstruction.
+When necessary, the customer can failover to any of the clusters which have the backup for high availability.
+
+In terms of the CAP theorem, Multi-cluster Replication adds high availability to Temporal's strongly consistent system.
```
Comment (wxing1292) [inline docs/server/multi-cluster.md:24]: this seems to claim C & A & P, which is not the case.

Comment (sw-yx) [reply docs/server/multi-cluster.md:24]: can you suggest better wording? original wording was "Cadence Cross DC is an AP (in terms of CAP)." - but [here](https://docs.temporal.io/docs/concepts/introduction/#modern-solution) we claim that it is CP

File: docs/server/multi-cluster.md [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+
+## Version
+
+A **version** is a concept in Multi-cluster Replication which describes the chronological order of events (per customer namespace).
```
Comment (wxing1292) [inline docs/server/multi-cluster.md:28]: remaining section of this doc is copy pasted without modification except `cadence` -> `temporal`?

**Review by wxing1292 — CHANGES_REQUESTED** [review]
```
In terms of the CAP theorem, Multi-cluster Replication adds high availability to Temporal's strongly consistent system.&nbsp;<br class="Apple-interchange-newline"
```

can you delete this line?

File: docs/server/multi-cluster.md (hunk 2 above)
Comment (sw-yx) [reply docs/server/multi-cluster.md:28]: yes and renamed Domain to cluster. basic port for now, just to make sure we have docs on this feature

File: docs/server/multi-cluster.md (hunk 1 above)
Comment (wxing1292) [reply docs/server/multi-cluster.md:24]: single DC setup is CP
cros DC setup is AP

Comment (wxing1292) [reply docs/server/multi-cluster.md:24]: i would just delete this line

Comment (sw-yx) [reply docs/server/multi-cluster.md:24]:
```suggestion
```

**Review by wxing1292 — APPROVED** [review]

**Review by rylandg — APPROVED** [review]
Overall looks great. Noticed that are still some references to DC:

https://deploy-preview-480--mystifying-fermi-1bc096.netlify.app/docs/server/namespaces/#what-is-the-recommended-pattern-to-send-external-events-to-an-active-cluster

Not sure if we care to change those but wanted to raise it.

**Comment by sw-yx** [comment]
test comment
<details><summary>Browser metadata</summary>





```
Path:      /
Browser:   Chrome 91.0.4472.114 on Mac OS 10.15.7
Viewport:  953 x 820 @2x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://live.browserstack.com/dashboard#url=https%3A%2F%2Fdeploy-preview-480--mystifying-fermi-1bc096.netlify.app%2F&os=OS+X&os_version=Catalina&browser=Chrome&browser_version=91.0&resolution=953x820&speed=1&start=true)



</details>

[Open Deploy Preview](https://deploy-preview-480--mystifying-fermi-1bc096.netlify.app/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=60cab7b6d046650007b85d15&commentID=60e885bd03eadf4d8d749c9a&resolution=resolved)

**Pull request merged**